docs/contracts/NFT Contract/NFT Actions/setprchsreq.a.md

```diff
@@ -49,6 +49,63 @@ If RAM price is greater than `maximum_uos_payment` transaction reverts.
 
 If token factory is inactive transaction reverts as well.
 
+`group_restriction` is an optional parameter that accepts a vector of 64-bit integers. This vector is designed to apply logical restrictions based on the group IDs that users belong to. The structure of each 64-bit integer is as follows:
+
+* The lower 60 bits represent the group ID.
+* The upper 4 bits are reserved for logical operators.
+
+Logical Operators
+The logical operators are defined as bitwise flags in the following manner:
+
+```c++
+#define OR        0X1000'0000'0000'0000   // 0: AND, 1: OR
+#define NEGATION  0X2000'0000'0000'0000   // 0: No negation, 1: Negation
+```
+
+* `4th Bit (OR Operator)`: If this bit is set, it indicates the OR operator. Otherwise, it defaults to the AND operator.
+* `3rd Bit (NEGATION Operator)`: If this bit is set, it indicates that the NEGATION should be applied to the group ID.
+
+The expression is evaluated from left to right, and parentheses are not used. Logical operators take into account starting from the second group ID in the sequence.
+
+For example, a `group_restriction` vector like `<OR + NOT + group1, AND + group2>` would be evaluated as `(NOT group1 AND group2)`. This means the condition will pass if the user is not a member of `group1` AND is a member of `group2`.
+
+Longer expression example for a `group_restriction` vector of `<OR + NOT + group1, AND + group2, OR + NOT + group3, OR + group4, AND + NOT + group5>`, the logical expression becomes:
+
+```css
+(NOT group1 AND group2) OR (NOT group3) OR (group4 AND NOT group5)
+```
+
+This means the condition will pass if:
+
+* The user is not a member of `group1` AND is a member of `group2`, OR
+* The user is not a member of `group3`, OR
+* The user is a member of `group4` AND not a member of `group5`.
+
+
+Formalization of Logical Expression Evaluation
+
+To formalize the evaluation process, let's consider the `group_restriction` vector as `G = [g1, g2, g3, ..., gn]`, where each `gi` is a 64-bit integer that encodes both the logical operator and the group ID.
+
+The corresponding logical expression `L` can be represented as:
+
+```scss
+L = O1(g1) OP2 O2(g2) OP3 O3(g3) ... OPn On(gn)
+
+```
+
+Where:
+
+* `Oi(gi)` represents the membership status for `gi`, which may be negated depending on the NEGATION bit.
```

Comment:
recommend to change to the following for clarity.

`Oi(gi)` checks whether a user belongs to the group `gi` if there is no NEGATION bit, otherwise, it checks whether a user **does not** belong to the group `gi`.